src/lib/support/BUILD.gn

```diff
@@ -73,6 +73,42 @@ source_set("attributes") {
   ]
 }
 
+source_set("logging_header") {
```

Comment:
So the story is:

1. CHIPLogging.cpp depends on Span.h
2. Span.h depends on CodeUtils.h
3. CodeUtils.h depends on CHIPLogging.h

If we want to keep CHIPLogging.h and CHIPLogging.cpp together, then all of these things need to go into the same source set. In addition to that it would need to contain CHIPMem.h (depended on by CHIPLogging.cpp) and we would have a dep on the platform logging stuff, because CHIPLogging.h uses `<platform/logging/LogV.h>`.  That might be an issue for things that just need to consume `Span` or might not....

Looks like this is all because Span.h uses VerifyOrDie, which can in fact log if CHIP_CONFIG_VERBOSE_VERIFY_OR_DIE....

So I could do some sort of source_set like that, and see how the deps fall out, but then what do we call it?  "core_support"?   "support_base"?  Something else?

Reply:
Per Slack discussion the plan is:

1. Have a source_set for "basic_logging" (name TBD) which has some new header name and new .cpp name which contain all the logging bits that do _not_ involve Span.
2. Have a source_set for CodeUtils.h (called "verifymacros" or TBD) that depends on "logging".
3. Have a source_set for "span" which depends on "verifymacros".
4. Have CHIPLogging depend on "basic_logging" and just include Span instead of forward-declaring it.

This breaks the circular dependency which is currently covered up by the forward-declares of Span that CHIPLogging.h has.